active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,565 @@
+
+
+The teaching tip has several states, including these notable ones.
+
+A teaching tip can point to a specific UI element to enhance contextual clarity of the information it is presenting. 
+
+![A sample app with a teaching tip pointing at the save button. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipPointerSample.jpg)
+
+A teaching tip can be used without a pointer when the information that it is presenting is not specific to a UI element. 
+
+![A sample app with a teaching tip in the bottom right corner. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipSampleApp.jpg)
+
+A teaching tip can require manual acknowledgement before dismissing or a teaching tip may also be light-dismiss enabled so that it will dismiss when a user scrolls or interacts with other elements of the application.
+
+A teaching tip can require the user to dismiss it via an "X" button in a top corner or a "Close" button at the bottom. A teaching tip may also be light-dismiss enabled in which case there is no dismiss button and the teaching tip will instead dismiss when a user scrolls or interacts with other elements of the application. Because of this behavior, light-dismiss tips are the best solution when a tip needs to be placed in a scrollable area. 
+
+![A sample app with a light-dismiss teaching tip in the bottom right corner. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to."](TeachingTipLightDismissSample.jpg)
+
+
+### Create a teaching tip
+
+Here's the XAML for a pointing teaching tip control that demonstrates the default look of the TeachingTip with a title and subtitle. 
+Note that the teaching tip can appear anywhere in the element tree. In this example below, it's located in a ResourceDictionary.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save">
+    <Button.Resources>
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Target="{x:Bind SaveButton}"
+            Title="Save automatically"
+            Subtitle="When you save your file to OneDrive, we save your changes as you go - so you never have to.">
+        </muxc:TeachingTip>
+    </Button.Resources>
+</Button>
+```
+
+C#
+```C#
+public MainPage()
+{
+    this.InitializeComponent();
+
+    if(!HaveExplainedAutoSave())
+    {
+        AutoSaveTip.IsOpen = true;
```

Comment:
Can I reconfigure the tip while it is open?

* Change title, subtitle, content, hero content, hero placement.
* Change target, placement, offset, root bounds constraint.
* Switch between light dismiss and explicit dismiss.
* Switch between "X" button and full Close button.
* Add/remove an action button.
* Change the content of the buttons.

Reply:
[Added a "Reconfiguring an open teaching tip" section to Remarks](https://github.com/Microsoft/microsoft-ui-xaml-specs/commit/b493afb1d70535508101aaeefaac1a922d3a7bcf).